packages/data-stores/src/user/index.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { registerStore } from '@wordpress/data';
+import wpcomProxyRequest from 'wpcom-proxy-request';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from './constants';
+import reducer, { State } from './reducer';
+import * as actions from './actions';
+import * as resolvers from './resolvers';
+import * as selectors from './selectors';
+import controls from './controls';
+import { DispatchFromMap, SelectFromMap } from '../mapped-types';
+
+export * from './types';
+export { State };
+
+const debug = debugFactory( 'data-stores:user' );
+
+let isRegistered = false;
+export function register(): typeof STORE_KEY {
```

Comment:
Silly me, `User.register()` can take parameters 😆 
I reckon pass the `client_id` and `client_secret` in here. Then you can use a control factory to create a controls that uses the correct config.

Reply:
Nice! Sounds great.